python-frontend/src/main/java/org/sonar/python/types/DeclaredType.java

```diff
@@ -81,7 +81,10 @@ public boolean canHaveMember(String memberName) {
 
   @Override
   public boolean declaresMember(String memberName) {
-    if (hasUnresolvedHierarchy()) {
+    Symbol fetchedTypeClass = getTypeClass();
+    if (hasUnresolvedHierarchy() ||
+      (fetchedTypeClass.fullyQualifiedName() != null &&
+        EXCLUDED_FQNS.contains(fetchedTypeClass.fullyQualifiedName()))) {
```

Comment:
I think the following
`EXCLUDED_FQNS.stream().anyMatch(this::mustBeOrExtend)`
would also include any subtypes of `Mock` and `MagicMock`, which could avoid other FPs for subtypes of mocks, such as:

```
class MyClassForMock(MagicMock):
    ...

def my_custom_mock(param: MyClassForMock):
    param + 3
    -param
```

Reply:
Thank you for the suggestion this really cleans up the code!